metagov/metagov/plugins/github/views.py

```diff
@@ -0,0 +1,57 @@
+import json
+import logging
+import requests
+
+from metagov.core.models import ProcessStatus
+from metagov.core.plugin_constants import AuthorizationType
+from metagov.plugins.github.models import Github, GithubIssueReactVote, GithubIssueCommentVote
+from metagov.plugins.github.utils import get_jwt
+
+logger = logging.getLogger(__name__)
+
+
+def process_event(request):
+
+    json_data = json.loads(request.body)
+
+    if "X-GitHub-Event" in request.headers:
+
+        for plugin in Github.objects.all():
+
+            if int(plugin.config["installation_id"]) == int(json_data["installation"]["id"]):
+
+                logger.info(f"Passing event to {plugin}")
+                plugin.github_webhook_receiver(request)
+
+                for process_type in [GithubIssueCommentVote, GithubIssueReactVote]:
+                    for process in process_type.objects.filter(plugin=plugin, status=ProcessStatus.PENDING.value):
+                        if hasattr(process, "receive_webhook"):
+                            process.receive_webhook(request)
+
+
+def get_authorize_url(state, type, community):
+    if type == AuthorizationType.APP_INSTALL:
+        return f"https://github.com/apps/metagovapp/installations/new/?state={state}"
+
+
+def auth_callback(type, code, redirect_uri, community, state, request):
```

Comment:
So I think if we DO want an access token for the user who installed, we would be able to get one by exchanging this `code` [like this](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#2-users-are-redirected-back-to-your-site-by-github). Maybe we don't need that yet, I'm not sure what scopes it provides that are different from the jwt token. Just noting that as a possibility in the future.

Reply:
Gotcha, yeah. For now I think it makes sense to just install it and act as an app, but if we do want either the installing user so we can act on their behalf, or just for efficiency's sake to use in the identity system, we know where to come back and make the change.